Scripts/Tracking/Velocity/VelocityEstimator.cs

```diff
@@ -191,12 +218,15 @@ protected virtual void ProcessEstimation()
         /// <param name="factor">The multiplier to apply to the transform difference.</param>
         protected virtual void EstimateVelocity(float factor)
         {
-            Vector3 currentPosition = (source != null ? source.transform.position : Vector3.zero);
+            Vector3 currentRelativePosition = relativeTo.TryGetPosition();
+            Vector3 relativeDeltaPosition = currentRelativePosition - previousRelativePosition;
+            Vector3 currentPosition = source.TryGetPosition();
             if (velocitySamples.Length > 0)

```

Comment:
Negate to `== 0` and create an early return with that. This way we don't do the math above this if without using the results.